org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/QuickAssistProcessor.java

```diff
@@ -4635,4 +4640,97 @@ private boolean getSplitSwitchLabelProposal(IInvocationContext context, ASTNode
 
 		return true;
 	}
+	private boolean getConvertFieldNamingConventionProposal(IInvocationContext context, ASTNode node, Collection<ICommandAccess> resultingCollections) {
+		if (!(node instanceof SimpleName)) {
+			return false;
+		}
+		SimpleName name= (SimpleName) node;
+		String selectedField = name.toString();
+		int offset = node.getStartPosition();
+		int length = node.getLength();
+		if (name.getAST().apiLevel() >= ASTHelper.JLS10 && name.isVar()) {
+			return false;
+		}
+
+		IEditorPart editor= ((AssistContext) context).getEditor();
+		if (!(editor instanceof JavaEditor))
+			return false;
+
+        @SuppressWarnings("unused")
+		List<String> fieldList = new ArrayList<String>();
+		CompilationUnit astRoot= context.getASTRoot();
+        TypeDeclaration typeDec = (TypeDeclaration) astRoot.types().get(0);
+        for(FieldDeclaration fd:typeDec.getFields()) {
+        	VariableDeclarationFragment fragment= (VariableDeclarationFragment) fd.fragments().get(0);
+			String fieldName= fragment.getName().getIdentifier();
+			int modifier = fd.getModifiers();
+			if(fieldName.equals(name.toString())) {
+				if(modifier!=24 && modifier!=25 && modifier!=26 && modifier!=28) {
```

Comment:
Don't use integers for modifier constants.  There are constants out there you can use.